### PR TITLE
Blob overminds can now request assistance from ghosts, who can then talk to them while they are orbiting them.

### PR DESCRIPTION
**Pull request by Mothblocks**
## About The Pull Request
~~Gives blob overminds sixth sense, AKA the ability to hear deadchat.~~
Blob overminds can now request assistance from ghosts, who can then talk to them while they are orbiting them.

https://file.house/Sdgs.mp4

## Why It's Good For The Game
A few reasons:

1. Takes more burden of knowledge off of blob, by letting you ask dead chat for advice without needing blobbernauts.
2. It's a ton of fun, I gave a blob sixth sense a round ago and everyone enjoyed it--it made things feel more hectic and made dead chat more engaged with the round.

If this is deemed too powerful, I can settle with making it a part of Distributed Neurons instead, though I don't think it's that strong. I do not want to make this a strain of its own, as I don't think it is powerful enough to where anyone would pick that.

## Changelog
:cl:
tweak: Blob overminds can now request assistance from ghosts, who can then talk to them while they are orbiting them.
/:cl:

### CODE REVIEW

**Comment by Pumpkinoe** [comment]
but what's the LOOORE.......

**Comment by Jared-Fogle** [comment]
its the sound of fallen blobbernauts

File: code/modules/antagonists/blob/overmind.dm [hunk 1]
```diff
@@ -63,6 +63,7 @@ GLOBAL_LIST_EMPTY(blob_nodes)
 		blob_core.update_icon()
 	SSshuttle.registerHostileEnvironment(src)
 	. = ..()
+	ADD_TRAIT(src, TRAIT_SIXTHSENSE, INNATE_TRAIT)
 	START_PROCESSING(SSobj, src)
```
Comment (MrDoomBringer) [inline code/modules/antagonists/blob/overmind.dm:68]: maybe add a short deadchat_broadcast that mentions to ghosts that the blob can communicate with them?

**Review by MrDoomBringer — CHANGES_REQUESTED** [review]

File: code/modules/antagonists/blob/overmind.dm (hunk 1 above)
Comment (Jared-Fogle) [reply code/modules/antagonists/blob/overmind.dm:68]: we didnt for revenant, theyll figure it out

Comment (ATH1909) [reply code/modules/antagonists/blob/overmind.dm:68]: add one for revenants too, then

**Comment by coiax** [comment]
This will lead to a little bit of strangeness for blobbernauts and sentient spores, since the overmind will be talking to ghosts, but they won't be able to hear dem ghosts.

**Comment by KathrinBailey** [comment]
I sure hope the blob doesn't have a secret weakness to salt, because deadchat will salt the blob to death in ten seconds.

**Comment by ishitbyabullet** [comment]
> 
> 
> I sure hope the blob doesn't have a secret weakness to salt, because deadchat will salt the blob to death in ten seconds.

blobs are just evolved snails.

File: code/modules/antagonists/blob/overmind.dm (hunk 1 above)
Comment (Jared-Fogle) [reply code/modules/antagonists/blob/overmind.dm:68]: people already figured it out, i don't think it's necessary

**Comment by wabash1** [comment]
Good PR, I was there for that round, I don't think blobbernauts or spores should hear ghosts as suggested, it would be harder to get them to listen having all that text flying by.

**Comment by Jared-Fogle** [comment]
> I don't think blobbernauts or spores should hear ghosts as suggested

They shouldn't, this is just the camera.

**Comment by IndieanaJones** [comment]
Kinda weird flavorwise but its not too much of a buff regardless.  Seems fine to me.

**Comment by Jared-Fogle** [comment]
flavor wont be an issue, blobs dont taste good anyway

**Comment by ishitbyabullet** [comment]
On second thought, it seems incredibly overpowered if you can reliably ask deadchat to be your eyes, ears, and brother taking the controller to play the game for you. You can already get advice technically from blob zombies, blobbernauts, and spores if you have the right blob type.

**Comment by wabash1** [comment]
Don't worry technically this PR has already been playtested with multiple
players. Could consider upping the delta by a few tiles though if balance
is a concern.

**Comment by Jared-Fogle** [comment]
> and brother taking the controller to play the game for you

More backseat gaming than anything else, really--if you choose to just blindly take their advice I believe you're not going to be at a direct advantage as someone who doesn't. Dead chat can be wrong, misguided, unfocused, and also just slow or give bad advice.

**Comment by Winterous** [comment]
Blob spawns are often made trivial by people not knowing what the hell they're doing and getting trampled all over, so being able to get advice is a great idea.

**Comment by ExcessiveUseOfCobblestone** [comment]
Any concerns about metagame?

I mean I guess if you have deadchat access it's no longer metagame but I imagine there is a reason you can only see as far as your blob can go, and having access to a non-curated deadchat (aka when the game sets it even when no admins are present vs when an admin is babysitting it for an cute event) might disrupt that especially if you have some IC friends you can rely on to give you good info unchecked.

All roles could use a deadchat aide, and some of them don't get the benefit of bringing those players in to communicate as well as help them win more directly. Just a bit uncomfortable with the idea ghost chat is synonymous with a helpdesk that can also freely talk about events on the station. We have ahelp, blob specifically has roles many ghosts can plop into, I don't really see the desire for this to be a regular occurrence.

**Comment by Jared-Fogle** [comment]
> We have ahelp, blob specifically has roles many ghosts can plop into

The downside of using adminhelp to ask for help with blob is simple, you are limited to the very limited number of admins online that are able to take your request, which is frequently zero, either because there's no admins, they're busy dealing with tickets, or they just flat out don't know blob well enough to feel justified giving advice--I know that I'd fall into the last option. Not that deadchat is all knowing, or even intelligent, but it's a much wider, and more *engaging*, net.

Blob's ghost roles are limited to not just *after* spawning, but to roles that take a significant amount of time to get, something a fresh blob player might not be capable of doing.

**Comment by Jared-Fogle** [comment]
As for the metagame concerns, I can see it, but from my perspective I just simply don't *think* it's powerful enough information.

**Comment by wabash1** [comment]
I would not think of it so much as metagaming because the use case will be
more likely for static help than meta. While you could say hey there's
people in medbay, if a blob is smack dab on the other side of the station
that does not help it much. More likely: hey blob you're new and those guys
coming from your left with laser guns repeatedly have rechargers which you
only wouldn't know if you were new, blob go east to this APC and take it
out to cut off their power temporarily. Most information the blob will get
will be map information, tactics, or a slightly early knowledge of a danger
like emitters. Worst case they get memed on by ghosts: hey blob expand
north there's no one there (while the clown is up there entertaining the
entire security force)

**Comment by ExcessiveUseOfCobblestone** [comment]
"theyre hitting the north side" and blob immediately goes to north side for some reason.

It doesnt really matter how much can be gained from the information when its the fact it's making it ok is more of the issue. The blob has restrictions to what it can see and hear, that is intended design and providing a relatively easy route to bypass that as a default mechanic seems incredibly odd.

If the issue is blobs need help, surely we can generate a solution that keeps it immersive and not open them to the chat room where people can freely talk about OOC and inround concepts the blob shouldn't be privvy to.

**Comment by Winterous** [comment]
Perhaps there could be a ghost role for "blob under-minds", who function essentially identically to the overmind, but have no mechanical interaction with anything. They exist purely to assist with information, and get prioritised for blobbernaut and spore zombie roles over ghosts.

Make it so there can only be like 2 at once, and since they aren't ghosts they aren't allowed to use metaknowledge anymore.

**Comment by wabash1** [comment]
Having difficulty believing a shapeless blob can only see where it focuses.
I would think it would know all that its body is near and then some,
sensing heat, vibrations, and sound in areas with atmosphere.

**Comment by ExcessiveUseOfCobblestone** [comment]
you dont have to believe it you can test it ingame yourself :)

**Comment by D4C-420** [comment]
How about instead of having 6th sense as an innate ability what if it was instead a new node type. Make it cost say 30 recourses (or the same to make a fragile blob tile a reflective blob tile). That way it allows more well versed players to not have to listen to dead chat, gives a new area for a blob to defend and the station to attack while making it not a free bee for the blob players who choose to use it.

**Comment by wabash1** [comment]
Good suggestion but I'd say if this is chosen, make it so new blob players
with less than x amount of blob time get the upgrade for free as to not
punish them. Same for if it is a strain.

**Comment by Jared-Fogle** [comment]
Making a new blob type/strain for it defeats part of the point which is releasing some of the burden of knowledge from blob. You won't have resources before you spawn, you might not even *know* that you have that option, and you certainly don't want to spend resources on it if you don't know what's a good investment.

**Comment by ExcessiveUseOfCobblestone** [comment]
to clarify, if you can make a mechanic inherit to all blobs that allows them to pull in help in a more immersive manner than by reading the entirety of deadchat I will be fine with it, otherwise I'm going to end up closing this.

Doesnt need to be a purchasable or upgrade or anything like that, im fine with the idea that ghosts can assist blobs but I want to be sure that its through an IC channel that is strictly about helping the blob ideally putting the players that opt in under the same viewing restrictions as the blob themselves.

**Review by ExcessiveUseOfCobblestone — CHANGES_REQUESTED** [review]
see above

**Comment by wabash1** [comment]
Would this be done as blob eyes (like ai, I could see bugs caused as this
seems more complicated to implement) or a way for ghosts to click blob
tiles and opt in to lock their movement to blob tiles only, effectively
staying as eyes until blob death or delta? Keeping them as ghosts and still
easily opt-in to roles such as blobbernaut and spores.

 Rule wise would chatter be frowned upon? Grief ghost/revenant can hear
ghosts and it's really fun having the dead chat cheer you on.

**Comment by ExcessiveUseOfCobblestone** [comment]
I was thinking just generating eye mobs that camera pan identical to the blob and can talk on blob chat, you could flavor it as an internal node or something.

How it is implemented i do not care so long as it meets the above requirement of making sure codewise/adminwise ghosts on this channel do not reveal info the blob would not have access to (IE someone way above them making an emitter setup). We do not police ghost info so unless the mob has like noclip or something (see griefghost) I would like to avoid giving roles pure 24/7 access to deadchat. This lets deadchat be kept as an open chat room where you can discuss anything in/out of the round you want (not to mention theres like literal admin indicators such as dsay and more recently event triggering broadcasted to deadchat now).

**Comment by Jared-Fogle** [comment]
> to clarify, if you can make a mechanic inherit to all blobs that allows them to pull in help in a more immersive manner than by reading the entirety of deadchat I will be fine with it

I'd like to see more clarification on what you mean by this, as I'm not sure why a blob would forego using this ability, especially since I've stated before why I wouldn't want to see it cost resources.

**Comment by Winterous** [comment]
> 
> 
> I was thinking just generating eye mobs that camera pan identical to the blob and can talk on blob chat, you could flavor it as an internal node or something.

This is what I was saying with the "undermind", a view-restricted, IC-knowledge-restricted helper.

**Comment by ExcessiveUseOfCobblestone** [comment]
> 
> 
> > to clarify, if you can make a mechanic inherit to all blobs that allows them to pull in help in a more immersive manner than by reading the entirety of deadchat I will be fine with it
> 
> I'd like to see more clarification on what you mean by this, as I'm not sure why a blob would forego using this ability, especially since I've stated before why I wouldn't want to see it cost resources.

it can be free, embedded into the role itself as opt-out instead of buying something to make it opt in, and applicable to all blob types.

Yes, if it was deadchat where you could benefit from it by getting otherwise metaknowledge, I would expect everyone to use it. For something that is strictly intended to guide new players, I would imagine there'd be an opt out option.

To clarify, my issue is with the blob accessing deadchat, not in what scenario the blob would receive the ability.

**Comment by Jared-Fogle** [comment]
The part I'm struggling with is how that's different from what this is other than adding a toggle for it, is it that the implication is admins will enforce people don't give the blob information they shouldn't?

**Comment by Winterous** [comment]
Yes. Dead chat is free game. This is an IC role with IC information restrictions.

**Comment by ExcessiveUseOfCobblestone** [comment]
If you would like the answer strictly to code/game end, it saves you the player from having to avoid discussing certain topics in deadchat if for some reason you dont want the blob to win (looks like you cant talk about how the humans are overcoming one side because the blob will just beeline their camera to that after you do).

From an admin pov, this means it can be curated if admins wished (they should). Maybe force-observe on the blob so the chat is exclusively how to help the blob through understanding game mechanics, not being a legal form of scouting/metacomming.

**Comment by Jared-Fogle** [comment]
Yeah, I was thinking something like you had to orbit the blob in order to talk to them.

**Comment by Jared-Fogle** [comment]
@ExcessiveUseOfCobblestone Let me know what you think: https://file.house/Sdgs.mp4

**Review by Twaticus — APPROVED** [review]

**Comment by Rohesie** [comment]
I'd prefer it if we used a feature such as the (already existing) imaginary friends.
That way some degree of OOC and IC separation can be enforced.

**Comment by Rohesie** [comment]
Closing this at the author's request, as they're busy with another PR and can't spare much time for changes here.

**Comment by MrDoomBringer** [comment]
this PR deserves life
ooc/ic....just call it voices of dead blobbernauts...

**Pull request closed without merging**